packages/backend/src/services/UnfurlService/UnfurlService.ts

```diff
@@ -492,15 +492,21 @@ export class UnfurlService extends BaseService {
                                     timeout: 60000,
                                 }); // NOTE: No await here
                             });
-                        } else if (
-                            lightdashPage === LightdashPage.CHART ||
-                            lightdashPage === LightdashPage.EXPLORE
-                        ) {
-                            // Wait for the visualization to load if we are in an explore page
+                        } else if (lightdashPage === LightdashPage.CHART) {
+                            // Wait for the visualization to load if we are in an saved explore page
                             const responsePattern = new RegExp(
                                 `${resourceUuid}/results`,
                             );
 
+                            chartResultsPromises = [
+                                page?.waitForResponse(responsePattern, {
+                                    timeout: 60000,
+                                }), // NOTE: No await here
+                            ];
+                        } else if (lightdashPage === LightdashPage.EXPLORE) {
```

Comment:
more elegant ways to handle this are welcome! 

We need to define the `waitForResponse`s like this and then await them after - playwright way.

Reply:
Seems pretty reasonable. Most alternatives I can think of would be much more intrusive.